<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<meta property="og:title" content="REFERENCES | Tidy Modeling with R" />
<meta property="og:type" content="book" />
<meta property="og:image" content="/images/cover.png" />
<meta property="og:description" content="The tidymodels framework is a collection of R packages for modeling and machine learning using tidyverse principles. This book provides a thorough introduction to how to use tidymodels, and an outline of good methodology and statistical practice for phases of the modeling process." />
<meta name="github-repo" content="tidymodels/TMwR" />

<meta name="author" content="Max Kuhn and Julia Silge" />


<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  TeX: { equationNumbers: { autoNumber: "AMS" } }
});
</script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<meta name="description" content="The tidymodels framework is a collection of R packages for modeling and machine learning using tidyverse principles. This book provides a thorough introduction to how to use tidymodels, and an outline of good methodology and statistical practice for phases of the modeling process.">

<title>REFERENCES | Tidy Modeling with R</title>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="libs/navigation-1.1/tabsets.js"></script>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
/* show arrow before summary tag as in bootstrap
TODO: remove if boostrap in updated in html_document (rmarkdown#1485) */
details > summary {
  display: list-item;
  cursor: pointer;
}
</style>
</head>

<body>

<div class="container-fluid main-container">


<div class="row">
<div class="col-sm-12">
<div id="TOC">
<ul>
<li class="has-sub"><a href="index.html#hello-world">Hello World</a>
<ul>
<li><a href="acknowledgments.html#acknowledgments">Acknowledgments</a></li>
<li><a href="using-code-examples.html#using-code-examples">Using Code Examples</a></li>
</ul></li>
<li class="part"><span><b>Introduction</b></span></li>
<li class="has-sub"><a href="1-software-modeling.html#software-modeling"><span class="toc-section-number">1</span> Software for modeling</a>
<ul>
<li><a href="1.1-fundamentals-for-modeling-software.html#fundamentals-for-modeling-software"><span class="toc-section-number">1.1</span> Fundamentals for Modeling Software</a></li>
<li class="has-sub"><a href="1.2-model-types.html#model-types"><span class="toc-section-number">1.2</span> Types of Models</a>
<ul>
<li><a href="1.2-model-types.html#descriptive-models">Descriptive models</a></li>
<li><a href="1.2-model-types.html#inferential-models">Inferential models</a></li>
<li><a href="1.2-model-types.html#predictive-models">Predictive models</a></li>
</ul></li>
<li><a href="1.3-connections-between-types-of-models.html#connections-between-types-of-models"><span class="toc-section-number">1.3</span> Connections Between Types of Models</a></li>
<li><a href="1.4-model-terminology.html#model-terminology"><span class="toc-section-number">1.4</span> Some Terminology</a></li>
<li><a href="1.5-model-phases.html#model-phases"><span class="toc-section-number">1.5</span> How Does Modeling Fit into the Data Analysis Process?</a></li>
<li><a href="1.6-software-summary.html#software-summary"><span class="toc-section-number">1.6</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="2-tidyverse.html#tidyverse"><span class="toc-section-number">2</span> A Tidyverse Primer</a>
<ul>
<li class="has-sub"><a href="2.1-tidyverse-principles.html#tidyverse-principles"><span class="toc-section-number">2.1</span> Tidyverse Principles</a>
<ul>
<li><a href="2.1-tidyverse-principles.html#design-for-humans"><span class="toc-section-number">2.1.1</span> Design for humans</a></li>
<li><a href="2.1-tidyverse-principles.html#reuse-existing-data-structures"><span class="toc-section-number">2.1.2</span> Reuse existing data structures</a></li>
<li><a href="2.1-tidyverse-principles.html#design-for-the-pipe-and-functional-programming"><span class="toc-section-number">2.1.3</span> Design for the pipe and functional programming</a></li>
</ul></li>
<li><a href="2.2-examples-of-tidyverse-syntax.html#examples-of-tidyverse-syntax"><span class="toc-section-number">2.2</span> Examples of Tidyverse Syntax</a></li>
<li><a href="2.3-chapter-summary.html#chapter-summary"><span class="toc-section-number">2.3</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="3-base-r.html#base-r"><span class="toc-section-number">3</span> A Review of R Modeling Fundamentals</a>
<ul>
<li><a href="3.1-an-example.html#an-example"><span class="toc-section-number">3.1</span> An Example</a></li>
<li><a href="3.2-formula.html#formula"><span class="toc-section-number">3.2</span> What Does the R Formula Do?</a></li>
<li><a href="3.3-tidiness-modeling.html#tidiness-modeling"><span class="toc-section-number">3.3</span> Why Tidiness is Important for Modeling</a></li>
<li><a href="3.4-combining-base-r-models-and-the-tidyverse.html#combining-base-r-models-and-the-tidyverse"><span class="toc-section-number">3.4</span> Combining Base R Models and the Tidyverse</a></li>
<li><a href="3.5-the-tidymodels-metapackage.html#the-tidymodels-metapackage"><span class="toc-section-number">3.5</span> The tidymodels Metapackage</a></li>
<li><a href="3.6-chapter-summary-1.html#chapter-summary-1"><span class="toc-section-number">3.6</span> Chapter Summary</a></li>
</ul></li>
<li class="part"><span><b>Modeling Basics</b></span></li>
<li class="has-sub"><a href="4-ames.html#ames"><span class="toc-section-number">4</span> The Ames Housing Data</a>
<ul>
<li><a href="4.1-exploring-features-of-homes-in-ames.html#exploring-features-of-homes-in-ames"><span class="toc-section-number">4.1</span> Exploring Features of Homes in Ames</a></li>
<li><a href="4.2-ames-summary.html#ames-summary"><span class="toc-section-number">4.2</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="5-splitting.html#splitting"><span class="toc-section-number">5</span> Spending our Data</a>
<ul>
<li><a href="5.1-splitting-methods.html#splitting-methods"><span class="toc-section-number">5.1</span> Common Methods for Splitting Data</a></li>
<li><a href="5.2-what-about-a-validation-set.html#what-about-a-validation-set"><span class="toc-section-number">5.2</span> What About a Validation Set?</a></li>
<li><a href="5.3-multi-level-data.html#multi-level-data"><span class="toc-section-number">5.3</span> Multi-Level Data</a></li>
<li><a href="5.4-other-considerations-for-a-data-budget.html#other-considerations-for-a-data-budget"><span class="toc-section-number">5.4</span> Other Considerations for a Data Budget</a></li>
<li><a href="5.5-splitting-summary.html#splitting-summary"><span class="toc-section-number">5.5</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="6-models.html#models"><span class="toc-section-number">6</span> Fitting Models with parsnip</a>
<ul>
<li><a href="6.1-create-a-model.html#create-a-model"><span class="toc-section-number">6.1</span> Create a Model</a></li>
<li><a href="6.2-use-the-model-results.html#use-the-model-results"><span class="toc-section-number">6.2</span> Use the Model Results</a></li>
<li><a href="6.3-parsnip-predictions.html#parsnip-predictions"><span class="toc-section-number">6.3</span> Make Predictions</a></li>
<li><a href="6.4-parsnip-extension-packages.html#parsnip-extension-packages"><span class="toc-section-number">6.4</span> parsnip-Extension Packages</a></li>
<li><a href="6.5-parsnip-addin.html#parsnip-addin"><span class="toc-section-number">6.5</span> Creating Model Specifications</a></li>
<li><a href="6.6-models-summary.html#models-summary"><span class="toc-section-number">6.6</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="7-workflows.html#workflows"><span class="toc-section-number">7</span> A Model Workflow</a>
<ul>
<li><a href="7.1-begin-model-end.html#begin-model-end"><span class="toc-section-number">7.1</span> Where Does the Model Begin and End?</a></li>
<li><a href="7.2-workflow-basics.html#workflow-basics"><span class="toc-section-number">7.2</span> Workflow Basics</a></li>
<li><a href="7.3-adding-raw-variables-to-the-workflow.html#adding-raw-variables-to-the-workflow"><span class="toc-section-number">7.3</span> Adding Raw Variables to the <code>workflow()</code></a></li>
<li class="has-sub"><a href="7.4-workflow-encoding.html#workflow-encoding"><span class="toc-section-number">7.4</span> How Does a <code>workflow()</code> Use the Formula?</a>
<ul>
<li><a href="7.4-workflow-encoding.html#tree-based-models">Tree-based models</a></li>
<li><a href="7.4-workflow-encoding.html#special-model-formulas"><span class="toc-section-number">7.4.1</span> Special formulas and in-line functions</a></li>
</ul></li>
<li><a href="7.5-workflow-sets-intro.html#workflow-sets-intro"><span class="toc-section-number">7.5</span> Creating Multiple Workflows at Once</a></li>
<li><a href="7.6-evaluating-the-test-set.html#evaluating-the-test-set"><span class="toc-section-number">7.6</span> Evaluating the Test Set</a></li>
<li><a href="7.7-workflows-summary.html#workflows-summary"><span class="toc-section-number">7.7</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="8-recipes.html#recipes"><span class="toc-section-number">8</span> Feature Engineering with recipes</a>
<ul>
<li><a href="8.1-a-simple-recipe-for-the-ames-housing-data.html#a-simple-recipe-for-the-ames-housing-data"><span class="toc-section-number">8.1</span> A Simple <code>recipe()</code> for the Ames Housing Data</a></li>
<li><a href="8.2-using-recipes.html#using-recipes"><span class="toc-section-number">8.2</span> Using Recipes</a></li>
<li><a href="8.3-how-data-are-used-by-the-recipe.html#how-data-are-used-by-the-recipe"><span class="toc-section-number">8.3</span> How Data are Used by the <code>recipe()</code></a></li>
<li class="has-sub"><a href="8.4-example-steps.html#example-steps"><span class="toc-section-number">8.4</span> Examples of <code>recipe()</code> Steps</a>
<ul>
<li><a href="8.4-example-steps.html#dummies"><span class="toc-section-number">8.4.1</span> Encoding qualitative data in a numeric format</a></li>
<li><a href="8.4-example-steps.html#interaction-terms"><span class="toc-section-number">8.4.2</span> Interaction terms</a></li>
<li><a href="8.4-example-steps.html#spline-functions"><span class="toc-section-number">8.4.3</span> Spline functions</a></li>
<li><a href="8.4-example-steps.html#feature-extraction"><span class="toc-section-number">8.4.4</span> Feature extraction</a></li>
<li><a href="8.4-example-steps.html#row-sampling-steps"><span class="toc-section-number">8.4.5</span> Row sampling steps</a></li>
<li><a href="8.4-example-steps.html#general-transformations"><span class="toc-section-number">8.4.6</span> General transformations</a></li>
<li><a href="8.4-example-steps.html#natural-language-processing"><span class="toc-section-number">8.4.7</span> Natural language processing</a></li>
</ul></li>
<li><a href="8.5-skip-equals-true.html#skip-equals-true"><span class="toc-section-number">8.5</span> Skipping Steps for New Data</a></li>
<li><a href="8.6-tidy-a-recipe.html#tidy-a-recipe"><span class="toc-section-number">8.6</span> Tidy a <code>recipe()</code></a></li>
<li><a href="8.7-column-roles.html#column-roles"><span class="toc-section-number">8.7</span> Column Roles</a></li>
<li><a href="8.8-recipes-summary.html#recipes-summary"><span class="toc-section-number">8.8</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="9-performance.html#performance"><span class="toc-section-number">9</span> Judging Model Effectiveness</a>
<ul>
<li><a href="9.1-performance-metrics-and-inference.html#performance-metrics-and-inference"><span class="toc-section-number">9.1</span> Performance Metrics and Inference</a></li>
<li><a href="9.2-regression-metrics.html#regression-metrics"><span class="toc-section-number">9.2</span> Regression Metrics</a></li>
<li><a href="9.3-binary-classification-metrics.html#binary-classification-metrics"><span class="toc-section-number">9.3</span> Binary Classification Metrics</a></li>
<li><a href="9.4-multi-class-classification-metrics.html#multi-class-classification-metrics"><span class="toc-section-number">9.4</span> Multi-Class Classification Metrics</a></li>
<li><a href="9.5-performance-summary.html#performance-summary"><span class="toc-section-number">9.5</span> Chapter Summary</a></li>
</ul></li>
<li class="part"><span><b>Tools for Creating Effective Models</b></span></li>
<li class="has-sub"><a href="10-resampling.html#resampling"><span class="toc-section-number">10</span> Resampling for Evaluating Performance</a>
<ul>
<li><a href="10.1-resampling-resubstition.html#resampling-resubstition"><span class="toc-section-number">10.1</span> The Resubstitution Approach</a></li>
<li class="has-sub"><a href="10.2-resampling-methods.html#resampling-methods"><span class="toc-section-number">10.2</span> Resampling Methods</a>
<ul>
<li><a href="10.2-resampling-methods.html#cv"><span class="toc-section-number">10.2.1</span> Cross-validation</a></li>
<li><a href="10.2-resampling-methods.html#repeated-cross-validation">Repeated cross-validation</a></li>
<li><a href="10.2-resampling-methods.html#leave-one-out-cross-validation">Leave-one-out cross-validation</a></li>
<li><a href="10.2-resampling-methods.html#monte-carlo-cross-validation">Monte Carlo cross-validation</a></li>
<li><a href="10.2-resampling-methods.html#validation"><span class="toc-section-number">10.2.2</span> Validation sets</a></li>
<li><a href="10.2-resampling-methods.html#bootstrap"><span class="toc-section-number">10.2.3</span> Bootstrapping</a></li>
<li><a href="10.2-resampling-methods.html#rolling"><span class="toc-section-number">10.2.4</span> Rolling forecasting origin resampling</a></li>
</ul></li>
<li><a href="10.3-resampling-performance.html#resampling-performance"><span class="toc-section-number">10.3</span> Estimating Performance</a></li>
<li><a href="10.4-parallel.html#parallel"><span class="toc-section-number">10.4</span> Parallel Processing</a></li>
<li><a href="10.5-extract.html#extract"><span class="toc-section-number">10.5</span> Saving the Resampled Objects</a></li>
<li><a href="10.6-resampling-summary.html#resampling-summary"><span class="toc-section-number">10.6</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="11-compare.html#compare"><span class="toc-section-number">11</span> Comparing Models with Resampling</a>
<ul>
<li><a href="11.1-workflow-set.html#workflow-set"><span class="toc-section-number">11.1</span> Creating Multiple Models with Workflow Sets</a></li>
<li><a href="11.2-resampled-stats.html#resampled-stats"><span class="toc-section-number">11.2</span> Comparing Resampled Performance Statistics</a></li>
<li><a href="11.3-simple-hypothesis-testing-methods.html#simple-hypothesis-testing-methods"><span class="toc-section-number">11.3</span> Simple Hypothesis Testing Methods</a></li>
<li class="has-sub"><a href="11.4-tidyposterior.html#tidyposterior"><span class="toc-section-number">11.4</span> Bayesian Methods</a>
<ul>
<li><a href="11.4-tidyposterior.html#the-effect-of-the-amount-of-resampling">The effect of the amount of resampling</a></li>
</ul></li>
<li><a href="11.5-compare-summary.html#compare-summary"><span class="toc-section-number">11.5</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="12-tuning.html#tuning"><span class="toc-section-number">12</span> Model Tuning and the Dangers of Overfitting</a>
<ul>
<li><a href="12.1-model-parameters.html#model-parameters"><span class="toc-section-number">12.1</span> Model Parameters</a></li>
<li><a href="12.2-tuning-parameter-examples.html#tuning-parameter-examples"><span class="toc-section-number">12.2</span> Tuning Parameters for Different Types of Models</a></li>
<li><a href="12.3-what-to-optimize.html#what-to-optimize"><span class="toc-section-number">12.3</span> What do we Optimize?</a></li>
<li><a href="12.4-overfitting-bad.html#overfitting-bad"><span class="toc-section-number">12.4</span> The consequences of poor parameter estimates</a></li>
<li><a href="12.5-two-general-strategies-for-optimization.html#two-general-strategies-for-optimization"><span class="toc-section-number">12.5</span> Two general strategies for optimization</a></li>
<li><a href="12.6-tuning-params-tidymodels.html#tuning-params-tidymodels"><span class="toc-section-number">12.6</span> Tuning Parameters in tidymodels</a></li>
<li><a href="12.7-chapter-summary-2.html#chapter-summary-2"><span class="toc-section-number">12.7</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="13-grid-search.html#grid-search"><span class="toc-section-number">13</span> Grid Search</a>
<ul>
<li class="has-sub"><a href="13.1-grids.html#grids"><span class="toc-section-number">13.1</span> Regular and Non-Regular Grids</a>
<ul>
<li><a href="13.1-grids.html#regular-grids">Regular grids</a></li>
<li><a href="13.1-grids.html#irregular-grids">Irregular grids</a></li>
</ul></li>
<li><a href="13.2-evaluating-grid.html#evaluating-grid"><span class="toc-section-number">13.2</span> Evaluating the Grid</a></li>
<li><a href="13.3-finalizing-the-model.html#finalizing-the-model"><span class="toc-section-number">13.3</span> Finalizing the Model</a></li>
<li><a href="13.4-tuning-usemodels.html#tuning-usemodels"><span class="toc-section-number">13.4</span> Tools for Creating Tuning Specifications</a></li>
<li class="has-sub"><a href="13.5-efficient-grids.html#efficient-grids"><span class="toc-section-number">13.5</span> Tools for Efficient Grid Search</a>
<ul>
<li><a href="13.5-efficient-grids.html#submodel-trick"><span class="toc-section-number">13.5.1</span> Submodel optimization</a></li>
<li><a href="13.5-efficient-grids.html#parallel-processing"><span class="toc-section-number">13.5.2</span> Parallel processing</a></li>
<li><a href="13.5-efficient-grids.html#benchmarking-boosted-trees"><span class="toc-section-number">13.5.3</span> Benchmarking boosted trees</a></li>
<li><a href="13.5-efficient-grids.html#access-to-global-variables"><span class="toc-section-number">13.5.4</span> Access to global variables</a></li>
<li><a href="13.5-efficient-grids.html#racing"><span class="toc-section-number">13.5.5</span> Racing methods</a></li>
</ul></li>
<li><a href="13.6-grid-summary.html#grid-summary"><span class="toc-section-number">13.6</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="14-iterative-search.html#iterative-search"><span class="toc-section-number">14</span> Iterative Search</a>
<ul>
<li><a href="14.1-svm.html#svm"><span class="toc-section-number">14.1</span> A Support Vector Machine Model</a></li>
<li class="has-sub"><a href="14.2-bayesian-optimization.html#bayesian-optimization"><span class="toc-section-number">14.2</span> Bayesian Optimization</a>
<ul>
<li><a href="14.2-bayesian-optimization.html#a-gaussian-process-model"><span class="toc-section-number">14.2.1</span> A Gaussian process model</a></li>
<li><a href="14.2-bayesian-optimization.html#acquisition-functions"><span class="toc-section-number">14.2.2</span> Acquisition functions</a></li>
<li><a href="14.2-bayesian-optimization.html#tune-bayes"><span class="toc-section-number">14.2.3</span> The <code>tune_bayes()</code> function</a></li>
</ul></li>
<li class="has-sub"><a href="14.3-simulated-annealing.html#simulated-annealing"><span class="toc-section-number">14.3</span> Simulated Annealing</a>
<ul>
<li><a href="14.3-simulated-annealing.html#simulated-annealing-search-process"><span class="toc-section-number">14.3.1</span> Simulated annealing search process</a></li>
<li><a href="14.3-simulated-annealing.html#tune-sim-anneal"><span class="toc-section-number">14.3.2</span> The <code>tune_sim_anneal()</code> function</a></li>
</ul></li>
<li><a href="14.4-iterative-summary.html#iterative-summary"><span class="toc-section-number">14.4</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="15-workflow-sets.html#workflow-sets"><span class="toc-section-number">15</span> Screening Many Models</a>
<ul>
<li><a href="15.1-modeling-concrete-mixture-strength.html#modeling-concrete-mixture-strength"><span class="toc-section-number">15.1</span> Modeling Concrete Mixture Strength</a></li>
<li><a href="15.2-creating-the-workflow-set.html#creating-the-workflow-set"><span class="toc-section-number">15.2</span> Creating the Workflow Set</a></li>
<li><a href="15.3-tuning-and-evaluating-the-models.html#tuning-and-evaluating-the-models"><span class="toc-section-number">15.3</span> Tuning and Evaluating the Models</a></li>
<li><a href="15.4-racing-example.html#racing-example"><span class="toc-section-number">15.4</span> Efficiently Screening Models</a></li>
<li><a href="15.5-finalizing-a-model.html#finalizing-a-model"><span class="toc-section-number">15.5</span> Finalizing a Model</a></li>
<li><a href="15.6-workflow-sets-summary.html#workflow-sets-summary"><span class="toc-section-number">15.6</span> Chapter Summary</a></li>
</ul></li>
<li class="part"><span><b>Beyond the Basics</b></span></li>
<li class="has-sub"><a href="16-dimensionality.html#dimensionality"><span class="toc-section-number">16</span> Dimensionality Reduction</a>
<ul>
<li><a href="16.1-when-problems-can-dimensionality-reduction-solve.html#when-problems-can-dimensionality-reduction-solve"><span class="toc-section-number">16.1</span> When Problems Can Dimensionality Reduction Solve?</a></li>
<li><a href="16.2-beans.html#beans"><span class="toc-section-number">16.2</span> A Picture is Worth a Thousand… Beans</a></li>
<li><a href="16.3-a-starter-recipe.html#a-starter-recipe"><span class="toc-section-number">16.3</span> A Starter Recipe</a></li>
<li class="has-sub"><a href="16.4-recipe-functions.html#recipe-functions"><span class="toc-section-number">16.4</span> Recipes in the Wild</a>
<ul>
<li><a href="16.4-recipe-functions.html#prep"><span class="toc-section-number">16.4.1</span> Preparing a recipe</a></li>
<li><a href="16.4-recipe-functions.html#bake"><span class="toc-section-number">16.4.2</span> Baking the recipe</a></li>
</ul></li>
<li class="has-sub"><a href="16.5-feature-extraction-techniques.html#feature-extraction-techniques"><span class="toc-section-number">16.5</span> Feature Extraction Techniques</a>
<ul>
<li><a href="16.5-feature-extraction-techniques.html#principal-component-analysis"><span class="toc-section-number">16.5.1</span> Principal component analysis</a></li>
<li><a href="16.5-feature-extraction-techniques.html#partial-least-squares"><span class="toc-section-number">16.5.2</span> Partial least squares</a></li>
<li><a href="16.5-feature-extraction-techniques.html#independent-component-analysis"><span class="toc-section-number">16.5.3</span> Independent component analysis</a></li>
<li><a href="16.5-feature-extraction-techniques.html#uniform-manifold-approximation-and-projection"><span class="toc-section-number">16.5.4</span> Uniform manifold approximation and projection</a></li>
</ul></li>
<li><a href="16.6-bean-models.html#bean-models"><span class="toc-section-number">16.6</span> Modeling</a></li>
<li><a href="16.7-dimensionality-summary.html#dimensionality-summary"><span class="toc-section-number">16.7</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="17-categorical.html#categorical"><span class="toc-section-number">17</span> Encoding Categorical Data</a>
<ul>
<li><a href="17.1-is-an-encoding-necessary.html#is-an-encoding-necessary"><span class="toc-section-number">17.1</span> Is an Encoding Necessary?</a></li>
<li><a href="17.2-encoding-ordinal-predictors.html#encoding-ordinal-predictors"><span class="toc-section-number">17.2</span> Encoding Ordinal Predictors</a></li>
<li class="has-sub"><a href="17.3-using-the-outcome-for-encoding-predictors.html#using-the-outcome-for-encoding-predictors"><span class="toc-section-number">17.3</span> Using the Outcome for Encoding Predictors</a>
<ul>
<li><a href="17.3-using-the-outcome-for-encoding-predictors.html#effect-encodings-with-partial-pooling"><span class="toc-section-number">17.3.1</span> Effect encodings with partial pooling</a></li>
</ul></li>
<li><a href="17.4-feature-hashing.html#feature-hashing"><span class="toc-section-number">17.4</span> Feature Hashing</a></li>
<li><a href="17.5-more-encoding-options.html#more-encoding-options"><span class="toc-section-number">17.5</span> More Encoding Options</a></li>
<li><a href="17.6-categorical-summary.html#categorical-summary"><span class="toc-section-number">17.6</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="18-explain.html#explain"><span class="toc-section-number">18</span> Explaining Models and Predictions</a>
<ul>
<li><a href="18.1-software-for-model-explanations.html#software-for-model-explanations"><span class="toc-section-number">18.1</span> Software for Model Explanations</a></li>
<li><a href="18.2-local-explanations.html#local-explanations"><span class="toc-section-number">18.2</span> Local Explanations</a></li>
<li><a href="18.3-global-explanations.html#global-explanations"><span class="toc-section-number">18.3</span> Global Explanations</a></li>
<li><a href="18.4-building-global-explanations-from-local-explanations.html#building-global-explanations-from-local-explanations"><span class="toc-section-number">18.4</span> Building Global Explanations from Local Explanations</a></li>
<li><a href="18.5-back-to-beans.html#back-to-beans"><span class="toc-section-number">18.5</span> Back to Beans!</a></li>
<li><a href="18.6-explain-summary.html#explain-summary"><span class="toc-section-number">18.6</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="19-trust.html#trust"><span class="toc-section-number">19</span> When Should You Trust Your Predictions?</a>
<ul>
<li><a href="19.1-equivocal-zones.html#equivocal-zones"><span class="toc-section-number">19.1</span> Equivocal Results</a></li>
<li><a href="19.2-applicability-domains.html#applicability-domains"><span class="toc-section-number">19.2</span> Determining Model Applicability</a></li>
<li><a href="19.3-trust-summary.html#trust-summary"><span class="toc-section-number">19.3</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="20-ensembles.html#ensembles"><span class="toc-section-number">20</span> Ensembles of Models</a>
<ul>
<li><a href="20.1-data-stack.html#data-stack"><span class="toc-section-number">20.1</span> Creating the Training Set for Stacking</a></li>
<li><a href="20.2-blend-predictions.html#blend-predictions"><span class="toc-section-number">20.2</span> Blend the Predictions</a></li>
<li><a href="20.3-fit-members.html#fit-members"><span class="toc-section-number">20.3</span> Fit the Member Models</a></li>
<li><a href="20.4-test-set-results.html#test-set-results"><span class="toc-section-number">20.4</span> Test Set Results</a></li>
<li><a href="20.5-ensembles-summary.html#ensembles-summary"><span class="toc-section-number">20.5</span> Chapter Summary</a></li>
</ul></li>
<li class="has-sub"><a href="21-inferential.html#inferential"><span class="toc-section-number">21</span> Inferential Analysis</a>
<ul>
<li><a href="21.1-inference-for-count-data.html#inference-for-count-data"><span class="toc-section-number">21.1</span> Inference for Count Data</a></li>
<li><a href="21.2-comparisons-with-two-sample-tests.html#comparisons-with-two-sample-tests"><span class="toc-section-number">21.2</span> Comparisons with Two-Sample Tests</a></li>
<li><a href="21.3-log-linear-models.html#log-linear-models"><span class="toc-section-number">21.3</span> Log-Linear Models</a></li>
<li><a href="21.4-a-more-complex-model.html#a-more-complex-model"><span class="toc-section-number">21.4</span> A More Complex Model</a></li>
<li><a href="21.5-inference-options.html#inference-options"><span class="toc-section-number">21.5</span> More Inferential Analysis</a></li>
<li><a href="21.6-inference-summary.html#inference-summary"><span class="toc-section-number">21.6</span> Chapter Summary</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li><a href="A-pre-proc-table.html#pre-proc-table"><span class="toc-section-number">A</span> Recommended Preprocessing</a></li>
<li><a href="references.html#references">REFERENCES</a></li>
</ul>
</div>
</div>
</div>
<div class="row">
<div class="col-sm-12">
<div id="references" class="section level1 unnumbered">
<h1>REFERENCES</h1>

<div id="refs" class="references csl-bib-body hanging-indent">
<div class="csl-entry">
Abrams, B. 2003. <span>“The Pit of Success.”</span> <a href="https://blogs.msdn.microsoft.com/brada/2003/10/02/the-pit-of-success/" class="uri">https://blogs.msdn.microsoft.com/brada/2003/10/02/the-pit-of-success/</a>.
</div>
<div class="csl-entry">
Baggerly, K, and K Coombes. 2009. <span>“Deriving Chemosensitivity from Cell Lines: <span>F</span>orensic Bioinformatics and Reproducible Research in High-Throughput Biology.”</span> <em>The Annals of Applied Statistics</em> 3 (4): 1309–34.
</div>
<div class="csl-entry">
Bartley, E AND Schliep, M . AND Hanks. 2019. <span>“Identifying and Characterizing Extrapolation in Multivariate Response Data.”</span> <em>PLOS ONE</em> 14 (December): 1–20.
</div>
<div class="csl-entry">
Biecek, Przemyslaw, and Tomasz Burzykowski. 2021. <em><span>Explanatory Model Analysis</span></em>. Chapman; Hall/CRC, New York. <a href="https://ema.drwhy.ai/">https://ema.drwhy.ai/</a>.
</div>
<div class="csl-entry">
Bohachevsky, I, M Johnson, and M Stein. 1986. <span>“Generalized Simulated Annealing for Function Optimization.”</span> <em>Technometrics</em> 28 (3): 209–17.
</div>
<div class="csl-entry">
Bolstad, B. 2004. <em>Low-Level Analysis of High-Density Oligonucleotide Array Data: Background, Normalization and Summarization</em>. University of California, Berkeley.
</div>
<div class="csl-entry">
Box, GEP, W Hunter, and J Hunter. 2005. <em>Statistics for Experimenters: An Introduction to Design, Data Analysis, and Model Building</em>. Wiley.
</div>
<div class="csl-entry">
Bradley, R, and M Terry. 1952. <span>“Rank Analysis of Incomplete Block Designs: <span>I.</span> The Method of Paired Comparisons.”</span> <em>Biometrika</em> 39 (3/4): 324–45.
</div>
<div class="csl-entry">
Breiman, L. 1996a. <span>“Bagging Predictors.”</span> <em>Machine Learning</em> 24 (2): 123–40.
</div>
<div class="csl-entry">
———. 1996b. <span>“Stacked Regressions.”</span> <em>Machine Learning</em> 24 (1): 49–64.
</div>
<div class="csl-entry">
———. 2001a. <span>“Random Forests.”</span> <em>Machine Learning</em> 45 (1): 5–32.
</div>
<div class="csl-entry">
———. 2001b. <span>“Statistical Modeling: The Two Cultures.”</span> <em>Statistical Science</em> 16 (3): 199–231.
</div>
<div class="csl-entry">
Carlson, B. 2012. <span>“Putting Oncology Patients at Risk.”</span> <em>Biotechnology Healthcare</em> 9 (3): 17–21.
</div>
<div class="csl-entry">
Chambers, J. 1998. <em>Programming with Data: A Guide to the s Language</em>. Berlin, Heidelberg: Springer-Verlag.
</div>
<div class="csl-entry">
Chambers, J, and T Hastie, eds. 1992. <em>Statistical Models in s</em>. Boca Raton, FL: CRC Press, Inc.
</div>
<div class="csl-entry">
Claeskens, G. 2016. <span>“Statistical Model Choice.”</span> <em>Annual Review of Statistics and Its Application</em> 3: 233–56.
</div>
<div class="csl-entry">
Cleveland, W. 1979. <span>“Robust Locally Weighted Regression and Smoothing Scatterplots.”</span> <em>Journal of the American Statistical Association</em> 74 (368): 829–36.
</div>
<div class="csl-entry">
Craig–Schapiro, R, M Kuhn, C Xiong, E Pickering, J Liu, T Misko, R Perrin, et al. 2011. <span>“Multiplexed Immunoassay Panel Identifies Novel <span>CSF</span> Biomarkers for <span class="nocase">Alzheimer’s</span> Disease Diagnosis and Prognosis.”</span> <em>PLoS ONE</em> 6 (4): e18850.
</div>
<div class="csl-entry">
Cybenko, G. 1989. <span>“Approximation by Superpositions of a Sigmoidal Function.”</span> <em>Mathematics of Control, Signals and Systems</em> 2 (4): 303–14.
</div>
<div class="csl-entry">
Danowski, T, J Aarons, J Hydovitz, and J Wingert. 1970. <span>“Utility of Equivocal Glucose Tolerances.”</span> <em>Diabetes</em> 19 (7): 524–26.
</div>
<div class="csl-entry">
Davison, A, and D Hinkley. 1997. <em>Bootstrap Methods and Their Application</em>. Vol. 1. Cambridge university press.
</div>
<div class="csl-entry">
De Cock, D. 2011. <span>“<span>Ames, Iowa</span>: Alternative to the <span>Boston</span> Housing Data as an End of Semester Regression Project.”</span> <em>Journal of Statistics Education</em> 19 (3).
</div>
<div class="csl-entry">
Dobson, A. 1999. <em>An Introduction to Generalized Linear Models</em>. Chapman; Hall: Boca Raton.
</div>
<div class="csl-entry">
Durrleman, S, and R Simon. 1989. <span>“Flexible Regression Models with Cubic Splines.”</span> <em>Statistics in Medicine</em> 8 (5): 551–61.
</div>
<div class="csl-entry">
Faraway, J. 2016. <em>Extending the Linear Model with <span>R</span>: Generalized Linear, Mixed Effects and Nonparametric Regression Models</em>. CRC press.
</div>
<div class="csl-entry">
Fox, J. 2008. <em>Applied Regression Analysis and Generalized Linear Models</em>. Second. Thousand Oaks, <span>CA</span>: Sage.
</div>
<div class="csl-entry">
Frazier, R. 2018. <span>“A Tutorial on Bayesian Optimization.”</span> <a href="https://arxiv.org/abs/1807.02811">https://arxiv.org/abs/1807.02811</a>.
</div>
<div class="csl-entry">
Freund, Y, and R Schapire. 1997. <span>“A Decision-Theoretic Generalization of on-Line Learning and an Application to Boosting.”</span> <em>Journal of Computer and System Sciences</em> 55 (1): 119–39.
</div>
<div class="csl-entry">
Friedman, J. 1991. <span>“Multivariate Adaptive Regression Splines.”</span> <em>The Annals of Statistics</em> 19 (1): 1–141.
</div>
<div class="csl-entry">
———. 2001. <span>“Greedy Function Approximation: A Gradient Boosting Machine.”</span> <em>Annals of Statistics</em> 29 (5): 1189–1232.
</div>
<div class="csl-entry">
Friedman, J, T Hastie, and R Tibshirani. 2010. <span>“Regularization Paths for Generalized Linear Models via Coordinate Descent.”</span> <em>Journal of Statistical Software</em> 33 (1): 1.
</div>
<div class="csl-entry">
Geladi, P., and B Kowalski. 1986. <span>“Partial Least-Squares Regression: A Tutorial.”</span> <em>Analytica Chimica Acta</em> 185: 1–17.
</div>
<div class="csl-entry">
Gentleman, R, V Carey, W Huber, R Irizarry, and S Dudoit. 2005. <em>Bioinformatics and Computational Biology Solutions Using <span>R</span> and <span>B</span>ioconductor</em>. Berlin, Heidelberg: Springer-Verlag.
</div>
<div class="csl-entry">
Good, I. J. 1985. <span>“Weight of Evidence: A Brief Survey.”</span> <em>Bayesian Statistics</em> 2: 249–70.
</div>
<div class="csl-entry">
Goodfellow, I, Y Bengio, and A Courville. 2016. <em>Deep Learning</em>. MIT Press.
</div>
<div class="csl-entry">
Guo, Cheng, and Felix Berkhahn. 2016. <span>“Entity Embeddings of Categorical Variables.”</span> <a href="http://arxiv.org/abs/1604.06737">http://arxiv.org/abs/1604.06737</a>.
</div>
<div class="csl-entry">
Hand, D, and R Till. 2001. <span>“A Simple Generalisation of the Area Under the <span>ROC</span> Curve for Multiple Class Classification Problems.”</span> <em>Machine Learning</em> 45 (August): 171–86.
</div>
<div class="csl-entry">
Hill, A, P LaPan, Y Li, and S Haney. 2007. <span>“Impact of Image Segmentation on High-Content Screening Data Quality for <span>SK</span>-<span>BR</span>-3 Cells.”</span> <em>BMC Bioinformatics</em> 8 (1): 340.
</div>
<div class="csl-entry">
Ho, T. 1995. <span>“Random Decision Forests.”</span> In <em>Proceedings of 3rd International Conference on Document Analysis and Recognition</em>, 1:278–82. IEEE.
</div>
<div class="csl-entry">
Hosmer, D, and Sy Lemeshow. 2000. <em>Applied Logistic Regression</em>. New York: John Wiley; Sons.
</div>
<div class="csl-entry">
Hvitfeldt, E., and J. Silge. 2021. <em>Supervised Machine Learning for Text Analysis in r</em>. A Chapman &amp; Hall Book. CRC Press. <a href="https://smltar.com/">https://smltar.com/</a>.
</div>
<div class="csl-entry">
Hyndman, R, and G Athanasopoulos. 2018. <em>Forecasting: Principles and Practice</em>. OTexts.
</div>
<div class="csl-entry">
Ismay, C, and A Kim. 2021. <em>Statistical Inference via Data Science: A ModernDive into r and the Tidyverse</em>. Chapman; Hall/CRC. <a href="https://moderndive.com/">https://moderndive.com/</a>.
</div>
<div class="csl-entry">
Jaworska, J, N Nikolova-Jeliazkova, and T Aldenberg. 2005. <span>“QSAR Applicability Domain Estimation by Projection of the Training Set in Descriptor Space: A Review.”</span> <em>Alternatives to Laboratory Animals</em> 33 (5): 445–59.
</div>
<div class="csl-entry">
Johnson, D, P Eckart, N Alsamadisi, H Noble, C Martin, and R Spicer. 2018. <span>“Polar Auxin Transport Is Implicated in Vessel Differentiation and Spatial Patterning During Secondary Growth in Populus.”</span> <em>American Journal of Botany</em> 105 (2): 186–96.
</div>
<div class="csl-entry">
Joseph, V, E Gul, and S Ba. 2015. <span>“Maximum Projection Designs for Computer Experiments.”</span> <em>Biometrika</em> 102 (2): 371–80.
</div>
<div class="csl-entry">
Jungsu, K, D Basak, and D Holtzman. 2009. <span>“The Role of Apolipoprotein <span>E</span> in <span class="nocase">Alzheimer’s</span> Disease.”</span> <em>Neuron</em> 63 (3): 287–303.
</div>
<div class="csl-entry">
Kerleguer, A., J.-L. Koeck, M. Fabre, P. Gérôme, R. Teyssou, and V. Hervé. 2003. <span>“Use of Equivocal Zone in Interpretation of Results of the Amplified <span>Mycobacterium Tuberculosis</span> Direct Test for Diagnosis of Tuberculosis.”</span> <em>Journal of Clinical Microbiology</em> 41 (4): 1783–84.
</div>
<div class="csl-entry">
Kirkpatrick, S, D Gelatt, and M Vecchi. 1983. <span>“Optimization by Simulated Annealing.”</span> <em>Science</em> 220 (4598): 671–80.
</div>
<div class="csl-entry">
Koklu, M, and IA Ozkan. 2020. <span>“Multiclass Classification of Dry Beans Using Computer Vision and Machine Learning Techniques.”</span> <em>Computers and Electronics in Agriculture</em> 174: 105507.
</div>
<div class="csl-entry">
Krueger, T, D Panknin, and M Braun. 2015. <span>“Fast Cross-Validation via Sequential Testing.”</span> <em>Journal of Machine Learning Research</em> 16 (33): 1103–55.
</div>
<div class="csl-entry">
Kruschke, J, and T Liddell. 2018. <span>“The <span>Bayesian</span> New Statistics: Hypothesis Testing, Estimation, Meta-Analysis, and Power Analysis from a <span>Bayesian</span> Perspective.”</span> <em>Psychonomic Bulletin and Review</em> 25 (1): 178–206.
</div>
<div class="csl-entry">
Kuhn, Max. 2014. <span>“Futility Analysis in the Cross-Validation of Machine Learning Models.”</span> <a href="https://arxiv.org/abs/1405.6974">https://arxiv.org/abs/1405.6974</a>.
</div>
<div class="csl-entry">
Kuhn, M, and K Johnson. 2013. <em>Applied Predictive Modeling</em>. Springer.
</div>
<div class="csl-entry">
———. 2020. <em>Feature Engineering and Selection: A Practical Approach for Predictive Models</em>. CRC Press.
</div>
<div class="csl-entry">
Lambert, D. 1992. <span>“Zero-Inflated Poisson Regression, with an Application to Defects in Manufacturing.”</span> <em>Technometrics</em> 34 (1): 1–14.
</div>
<div class="csl-entry">
Littell, R, J Pendergast, and R Natarajan. 2000. <span>“Modelling Covariance Structure in the Analysis of Repeated Measures Data.”</span> <em>Statistics in Medicine</em> 19 (13): 1793–1819.
</div>
<div class="csl-entry">
Long, J. 1992. <span>“<span class="nocase">Measures of Sex Differences in Scientific Productivity*</span>.”</span> <em>Social Forces</em> 71 (1): 159–78.
</div>
<div class="csl-entry">
Lundberg, Scott M., and Su-In Lee. 2017. <span>“A Unified Approach to Interpreting Model Predictions.”</span> In <em>Proceedings of the 31st International Conference on Neural Information Processing Systems</em>, 4768–77. NIPS’17. Red Hook, NY, USA: Curran Associates Inc.
</div>
<div class="csl-entry">
Mangiafico, S. 2015. <span>“An <span>R</span> Companion for the Handbook of Biological Statistics.”</span> <a href="https://rcompanion.org/handbook/" class="uri">https://rcompanion.org/handbook/</a>.
</div>
<div class="csl-entry">
Maron, O, and A Moore. 1994. <span>“Hoeffding Races: Accelerating Model Selection Search for Classification and Function Approximation.”</span> In <em>Advances in Neural Information Processing Systems</em>, 59–66.
</div>
<div class="csl-entry">
McCullagh, P, and J Nelder. 1989. <em>Generalized Linear Models</em>. London: Chapman; Hall.
</div>
<div class="csl-entry">
McDonald, J. 2009. <em>Handbook of Biological Statistics</em>. Sparky House Publishing.
</div>
<div class="csl-entry">
McElreath, R. 2020. <em>Statistical Rethinking: A <span>Bayesian</span> Course with Examples in <span>R</span> and <span>Stan</span></em>. CRC press.
</div>
<div class="csl-entry">
McInnes, L, J Healy, and J Melville. 2020. <span>“UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction.”</span>
</div>
<div class="csl-entry">
McKay, M, R Beckman, and W Conover. 1979. <span>“A Comparison of Three Methods for Selecting Values of Input Variables in the Analysis of Output from a Computer Code.”</span> <em>Technometrics</em> 21 (2): 239–45.
</div>
<div class="csl-entry">
Micci-Barreca, Daniele. 2001. <span>“A Preprocessing Scheme for High-Cardinality Categorical Attributes in Classification and Prediction Problems.”</span> <em>SIGKDD Explor. Newsl.</em> 3 (1): 27–32. <a href="https://doi.org/10.1145/507533.507538">https://doi.org/10.1145/507533.507538</a>.
</div>
<div class="csl-entry">
Mingqiang, Y, K Kidiyo, and R Joseph. 2008. <span>“A Survey of Shape Feature Extraction Techniques.”</span> In <em>Pattern Recognition</em>, edited by PY Yin. Rijeka: IntechOpen. <a href="https://doi.org/10.5772/6237">https://doi.org/10.5772/6237</a>.
</div>
<div class="csl-entry">
Molnar, Christopher. 2020. <em><span>Interpretable Machine Learning</span></em>. lulu.com. <a href="https://christophm.github.io/interpretable-ml-book/">https://christophm.github.io/interpretable-ml-book/</a>.
</div>
<div class="csl-entry">
Mullahy, J. 1986. <span>“Specification and Testing of Some Modified Count Data Models.”</span> <em>Journal of Econometrics</em> 33 (3): 341–65.
</div>
<div class="csl-entry">
Netzeva, T, A Worth, T Aldenberg, R Benigni, M Cronin, P Gramatica, J Jaworska, et al. 2005. <span>“Current Status of Methods for Defining the Applicability Domain of (Quantitative) Structure-Activity Relationships: The Report and Recommendations of ECVAM Workshop 52.”</span> <em>Alternatives to Laboratory Animals</em> 33 (2): 155–73.
</div>
<div class="csl-entry">
Olsson, D, and L Nelson. 1975. <span>“The <span>N</span>elder-<span>M</span>ead Simplex Procedure for Function Minimization.”</span> <em>Technometrics</em> 17 (1): 45–51.
</div>
<div class="csl-entry">
Opitz, J, and S Burst. 2019. <span>“Macro F1 and Macro F1.”</span> <a href="https://arxiv.org/abs/1911.03347">https://arxiv.org/abs/1911.03347</a>.
</div>
<div class="csl-entry">
R Core Team. 2014. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="http://www.R-project.org/">http://www.R-project.org/</a>.
</div>
<div class="csl-entry">
Rasmussen, C, and C Williams. 2006. <em>Gaussian Processes for Machine Learning</em>. <em>Gaussian Processes for Machine Learning</em>. MIT Press.
</div>
<div class="csl-entry">
Santner, T, B Williams, W Notz, and B Williams. 2003. <em>The Design and Analysis of Computer Experiments</em>. Springer.
</div>
<div class="csl-entry">
Schmidberger, M, M Morgan, D Eddelbuettel, H Yu, L Tierney, and U Mansmann. 2009. <span>“State of the Art in Parallel Computing with <span>R</span>.”</span> <em>Journal of Statistical Software</em> 31 (1): 1–27. <a href="https://www.jstatsoft.org/v031/i01">https://www.jstatsoft.org/v031/i01</a>.
</div>
<div class="csl-entry">
Schulz, E, M Speekenbrink, and A Krause. 2018. <span>“A Tutorial on Gaussian Process Regression: Modelling, Exploring, and Exploiting Functions.”</span> <em>Journal of Mathematical Psychology</em> 85: 1–16.
</div>
<div class="csl-entry">
Shahriari, B., K. Swersky, Z. Wang, R. P. Adams, and N. de Freitas. 2016. <span>“Taking the Human Out of the Loop: A Review of Bayesian Optimization.”</span> <em>Proceedings of the IEEE</em> 104 (1): 148–75.
</div>
<div class="csl-entry">
Shewry, M, and H Wynn. 1987. <span>“Maximum Entropy Sampling.”</span> <em>Journal of Applied Statistics</em> 14 (2): 165–70.
</div>
<div class="csl-entry">
Shmueli, G. 2010. <span>“To Explain or to Predict?”</span> <em>Statistical Science</em> 25 (3): 289–310.
</div>
<div class="csl-entry">
Symons, S, and RG Fulcher. 1988. <span>“Determination of Wheat Kernel Morphological Variation by Digital Image Analysis: <span>I</span>. <span>Variation</span> in Eastern Canadian Milling Quality Wheats.”</span> <em>Journal of Cereal Science</em> 8 (3): 211–18.
</div>
<div class="csl-entry">
Thomas, R, and D Uminsky. 2020. <span>“The Problem with Metrics Is a Fundamental Problem for AI.”</span> <a href="https://arxiv.org/abs/2002.08512">https://arxiv.org/abs/2002.08512</a>.
</div>
<div class="csl-entry">
Tibshirani, Robert. 1996. <span>“Regression Shrinkage and Selection via the Lasso.”</span> <em>Journal of the Royal Statistical Society. Series B (Methodological)</em> 58 (1): 267–88. <a href="http://www.jstor.org/stable/2346178">http://www.jstor.org/stable/2346178</a>.
</div>
<div class="csl-entry">
Van Laarhoven, P, and E Aarts. 1987. <span>“Simulated Annealing.”</span> In <em>Simulated Annealing: Theory and Applications</em>, 7–15. Springer.
</div>
<div class="csl-entry">
Wasserstein, R, and N Lazar. 2016. <span>“The <span>ASA</span> Statement on p-Values: Context, Process, and pPurpose.”</span> <em>The American Statistician</em> 70 (2): 129–33.
</div>
<div class="csl-entry">
Weinberger, K, A Dasgupta, J Langford, A Smola, and J Attenberg. 2009. <span>“Feature Hashing for Large Scale Multitask Learning.”</span> In <em>Proceedings of the 26th Annual International Conference on Machine Learning</em>, 1113–20. ACM.
</div>
<div class="csl-entry">
Wickham, H. 2019. <em>Advanced r</em>. 2nd ed. Chapman &amp; Hall/CRC the r Series. Taylor &amp; Francis. <a href="https://doi.org/10.1201/9781351201315">https://doi.org/10.1201/9781351201315</a>.
</div>
<div class="csl-entry">
Wickham, H, M Averick, J Bryan, W Chang, L McGowan, R François, G Grolemund, et al. 2019. <span>“Welcome to the <span>Tidyverse</span>.”</span> <em>Journal of Open Source Software</em> 4 (43).
</div>
<div class="csl-entry">
Wickham, H, and G Grolemund. 2016. <em><span class="sans-serif">R</span> for Data Science: <span>I</span>mport, Tidy, Transform, Visualize, and Model Data</em>. O’Reilly Media, Inc.
</div>
<div class="csl-entry">
Wolpert, D. 1992. <span>“Stacked Generalization.”</span> <em>Neural Networks</em> 5 (2): 241–59.
</div>
<div class="csl-entry">
Wu, X, and Z Zhou. 2017. <span>“A Unified View of Multi-Label Performance Measures.”</span> In <em>International Conference on Machine Learning</em>, 3780–88.
</div>
<div class="csl-entry">
Wundervald, B, A Parnell, and K Domijan. 2020. <span>“Generalizing Gain Penalization for Feature Selection in Tree-Based Models.”</span> <a href="https://arxiv.org/abs/2006.07515">https://arxiv.org/abs/2006.07515</a>.
</div>
<div class="csl-entry">
Xu, Q, and Y Liang. 2001. <span>“<span>Monte Carlo</span> Cross Validation.”</span> <em>Chemometrics and Intelligent Laboratory Systems</em> 56 (1): 1–11.
</div>
<div class="csl-entry">
Yeo, I-K, and R Johnson. 2000. <span>“A New Family of Power Transformations to Improve Normality or Symmetry.”</span> <em>Biometrika</em> 87 (4): 954–59.
</div>
<div class="csl-entry">
Zeileis, A, C Kleiber, and S Jackman. 2008. <span>“Regression Models for Count Data in <span>R</span>.”</span> <em>Journal of Statistical Software</em> 27 (8): 1–25. <a href="https://www.jstatsoft.org/v027/i08">https://www.jstatsoft.org/v027/i08</a>.
</div>
<div class="csl-entry">
Zumel, Nina, and John Mount. 2019. <span>“Vtreat: A Data.frame Processor for Predictive Modeling.”</span> <a href="http://arxiv.org/abs/1611.09477">http://arxiv.org/abs/1611.09477</a>.
</div>
</div>
</div>








































<p style="text-align: center;">
<a href="A-pre-proc-table.html"><button class="btn btn-default">Previous</button></a>
</p>
</div>
</div>


</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

</body>
</html>
